ansible/roles/cphc-migration/templates/cphc_connection_check.sh

```diff
@@ -1,19 +1,42 @@
 #! /usr/bin/bash
+# alerts configuration
+MIN_FAILURES_FOR_ALERT=2
+ALERT_FREQUENCY_MINUTES=5
+STATUS_CHECK_FREQUENCY_SECONDS=5
+ALERT_MESSAGE="Unable to connect CPHC VPN"
+# end alerts configuration
 
 CONSECUTIVE_FAILURES=0
+LAST_ALERT_TIME=0
+
+slack_alert () {
+  curl -X POST -H 'Content-type: application/json' --data '{"text":"$1"}' {{ cphc_vpn.alerts_webhook_url }}
+}
+
+vpn_login () {
+  naclient login -profile {{ cphc_vpn.profile }} -user {{ cphc_vpn.username }} -password '{{ cphc_vpn.password }}' <<< y
+}
+
+echo_with_time () {
+  echo "$(date '+TIME:%H:%M:%S') $1"
+}
+
 while true; do
   naclient status
   if [ $? -eq 1 ]; then
-    echo "$(date '+TIME:%H:%M:%S') VPN is connected"
+    echo_with_time "VPN is connected"
   else
     CONSECUTIVE_FAILURES=$((CONSECUTIVE_FAILURES + 1))
-    naclient login -profile {{ cphc_vpn.profile }} -user {{ cphc_vpn.username }} -password '{{ cphc_vpn.password }}' <<< y
+    vpn_login
     if [ $? -eq 1 ]; then
-      echo "$(date '+TIME:%H:%M:%S') Successfully connected to VPN"
+      echo_with_time "Successfully connected to VPN"
       CONSECUTIVE_FAILURES=0
-    elif [ $CONSECUTIVE_FAILURES -gt 2 ]; then
-      curl -X POST -H 'Content-type: application/json' --data '{"text":"Unable to connect CPHC VPN"}' {{ cphc_vpn.alerts_webhook_url }}
+      LAST_ALERT_TIME=0
+    elif [ $CONSECUTIVE_FAILURES -gt $MIN_FAILURES_FOR_ALERT ] && [ $(($(date +%s) - $LAST_ALERT_TIME)) -ge $(( ALERT_FREQUENCY_MINUTES * 60 )) ]; then
```

Comment:
We're not resetting consecutive failures here, that's intentional right? Once we hit the threshold, we'll alert every check until resolved?

Reply:
Yeah, we'll keep alerting if there are consecutive alerts.